tasks/lzards-backup/src/index.ts

```diff
@@ -0,0 +1,277 @@
+      },
+    });
+};
+
+export const makeBackupFileRequest = async (params: {
+  authToken: string,
+  collection: string,
+  creds: AWS.STS.AssumeRoleResponse,
+  file: MessageGranuleFilesObject,
+  granuleId: string,
+  lzardsPostMethod?: typeof postRequestToLzards,
+  generateAccessUrlMethod?: typeof generateAccessUrl,
+}): Promise<makeBackupFileRequestResult> => {
+  const {
+    authToken,
+    collection,
+    creds,
+    file,
+    granuleId,
+    lzardsPostMethod = postRequestToLzards,
+    generateAccessUrlMethod = generateAccessUrl,
+  } = params;
+
+  const { Key, Bucket } = parseS3Uri(file.filename);
+
+  const accessUrl = await generateAccessUrlMethod({
+    Bucket,
+    creds,
+    Key,
+  });
+  log.info(`${granuleId}: posting backup request to LZARDS: ${file.filename}`);
+  try {
+    const { statusCode, body } = await lzardsPostMethod({
+      accessUrl,
+      authToken,
+      collection,
+      file,
+      granuleId,
+    });
+    if (statusCode !== 201) {
+      log.error(`${granuleId}: Request failed - LZARDS api returned ${statusCode}: ${JSON.stringify(body)}`);
+      return { statusCode, granuleId, filename: file.filename, body, status: 'FAILED' };
+    }
+    return { statusCode, granuleId, filename: file.filename, body, status: 'COMPLETED' };
+  } catch (error) {
+    log.error(`${granuleId}: LZARDS request failed: ${error}`);
+    return { granuleId, filename: file.filename, body: JSON.stringify(error), status: 'FAILED' };
+  }
+};
+
+export const shouldBackupFile = (
+  fileName: string,
+  collectionConfig: PartialCollectionRecord
+): boolean => {
+  const collectionFiles = collectionConfig?.files || [];
+  const config = collectionFiles.find(
+    ({ regex }) => fileName.match(regex)
+  );
+  if (config?.lzards?.backup) return true;
+  return false;
+};
+
+export const getGranuleCollection = async (params: {
+  collectionName: string,
+  collectionVersion: string,
+  stackPrefix?: string
+}): Promise<PartialCollectionRecord> => {
+  const prefix = params.stackPrefix || getRequiredEnvVar('stackName');
+  const { collectionName, collectionVersion } = params;
+  if (!collectionName && !collectionVersion) {
+    throw new CollectionError('Collection Name and Version not defined');
+  }
+  const collectionResults = await getCollections({
+    prefix,
+    query: { name: collectionName, version: collectionVersion },
+  });
+  return JSON.parse(collectionResults.body).results[0] as PartialCollectionRecord;
+};
+
+export const backupGranule = async (
+  creds: AWS.STS.AssumeRoleResponse,
+  authToken: string,
+  granule: MessageGranule
+) => {
+  log.info(`${granule.granuleId}: Backup called on granule: ${JSON.stringify(granule)}`);
+  let granuleCollection: PartialCollectionRecord;
+  try {
+    granuleCollection = await getGranuleCollection({
+      collectionName: granule.dataType,
+      collectionVersion: granule.version,
+    });
+  } catch (error) {
+    if (error.name === 'CollectionNotDefinedError') {
```

Comment:
It was, however this was missed on refactor. 

This will fail (and pass tests related) even if granuleCollection is undefined as `shouldBackupFile` will throw, but no point in relying on that, for obvious reasons. 

Refactored: 09cd89eaa

Reply:
09cd89eaa